Crash on start with siva files

```
 pga list -l python -f json | head -n 1 | jq -r '.sivaFilenames[]' | pga get -i -o repos

./gitbase server -v -d /tmp/repos/ -i /tmp/index

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x184cab4]

goroutine 1 [running]:
github.com/src-d/gitbase.readRefs(0x2035380, 0xc000a27410, 0x14, 0x20, 0x14, 0x0, 0x0)
	/github.com/src-d/gitbase/checksum.go:98 +0x204
github.com/src-d/gitbase.(*checksumable).Checksum(0xc000a27500, 0x0, 0x0, 0x0, 0x0)
	/github.com/src-d/gitbase/checksum.go:32 +0x1ce
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql.(*IndexRegistry).LoadIndexes(0xc000a19ae0, 0xc0004a7aa0, 0x2, 0x2, 0x0, 0x0)
	/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/index.go:233 +0x7c7
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server%2ev0.(*Engine).Init(0xc0004a7520, 0x1, 0x1)
	/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/engine.go:121 +0x43
github.com/src-d/gitbase/cmd/gitbase/command.(*Server).buildDatabase(0xc0001382a0, 0xc0004a74a0, 0xc00000f3c0)
	/github.com/src-d/gitbase/cmd/gitbase/command/server.go:225 +0x319
github.com/src-d/gitbase/cmd/gitbase/command.(*Server).Execute(0xc0001382a0, 0xc00026c420, 0x0, 0x6, 0x0, 0x0)
	/github.com/src-d/gitbase/cmd/gitbase/command/server.go:131 +0x23d
github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc00026c3c0, 0xc00003a080, 0x6, 0x6, 0xc000a248b0, 0x1, 0x2, 0xc00012c630, 0xc000480e00)
	/github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags/parser.go:316 +0x80b
github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags.(*Parser).Parse(0xc00026c3c0, 0x1d7f997, 0x7, 0x1e4787d, 0x1c, 0x1e4787d)
	/github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags/parser.go:186 +0x71
main.main()
	/github.com/src-d/gitbase/cmd/gitbase/main.go:52 +0x39d
```

--- a/checksum.go
+++ b/checksum.go
@@ -5,6 +5,8 @@ import (
 	"crypto/sha1"
 	"encoding/base64"
 	"io"
+	"sort"
+	"strings"
 
 	git "gopkg.in/src-d/go-git.v4"
 	"gopkg.in/src-d/go-git.v4/plumbing"
@@ -79,6 +81,24 @@ func readChecksum(r repository) ([]byte, error) {
 	return result, nil
 }
 
+type reference struct {
+	name string
+	hash string
+}
+
+type references []reference
+
+type byHashAndName []reference
+
+func (b byHashAndName) Len() int      { return len(b) }
+func (b byHashAndName) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byHashAndName) Less(i, j int) bool {
+	if cmp := strings.Compare(b[i].hash, b[j].hash); cmp != 0 {
+		return cmp < 0
+	}
+	return strings.Compare(b[i].name, b[j].name) < 0
+}
+
 func readRefs(r repository) ([]byte, error) {
 	repo, err := r.Repo()
 	if err != nil {
@@ -90,26 +110,28 @@ func readRefs(r repository) ([]byte, error) {
 
 	buf := bytes.NewBuffer(nil)
 
-	head, err := repo.Head()
-	if err != nil && err != plumbing.ErrReferenceNotFound {
-		return nil, err
-	} else {
-		buf.WriteString("HEAD")
-		buf.WriteString(head.Hash().String())
-	}
-
 	refs, err := repo.References()
 	if err != nil {
 		return nil, err
 	}
 
+	var references []reference
 	err = refs.ForEach(func(r *plumbing.Reference) error {
-		buf.WriteString(string(r.Name()))
-		buf.WriteString(r.Hash().String())
+		references = append(references, reference{
+			name: string(r.Name()),
+			hash: r.Hash().String(),
+		})
 		return nil
 	})
 	if err != nil {
 		return nil, err
+	}
+
+	sort.Stable(byHashAndName(references))
+
+	for _, r := range references {
+		buf.WriteString(r.name)
+		buf.WriteString(r.hash)
 	}
 
 	return buf.Bytes(), nil

--- a/checksum_test.go
+++ b/checksum_test.go
@@ -2,6 +2,8 @@ package gitbase
 
 import (
 	"fmt"
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -27,7 +29,7 @@ func TestChecksum(t *testing.T) {
 	c := &checksumable{pool}
 	checksum, err := c.Checksum()
 	require.NoError(err)
-	require.Equal("ogfv7HAwFigDgtuW4tbnEP+Zl40=", checksum)
+	require.Equal("mGPoKCyOIkXX4reGe1vTBPIOg2E=", checksum)
 
 	pool = NewRepositoryPool(cache.DefaultMaxSize)
 	path := fixtures.ByTag("worktree").One().Worktree().Root()
@@ -36,5 +38,53 @@ func TestChecksum(t *testing.T) {
 	c = &checksumable{pool}
 	checksum, err = c.Checksum()
 	require.NoError(err)
-	require.Equal("5kfLCygyBSZFMh+nFzFNk3zAUTQ=", checksum)
+	require.Equal("rwQnBj7HRazv9wuU//nQ+nuf0WY=", checksum)
+}
+
+func TestChecksumSiva(t *testing.T) {
+	require := require.New(t)
+
+	pool := NewRepositoryPool(cache.DefaultMaxSize)
+	require.NoError(
+		filepath.Walk("_testdata", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if IsSivaFile(path) {
+				require.NoError(pool.AddSivaFile(path))
+			}
+
+			return nil
+		}),
+	)
+
+	c := &checksumable{pool}
+	checksum, err := c.Checksum()
+	require.NoError(err)
+	require.Equal("wJEvZNAc7QRszsf9KhGu+UeKto0=", checksum)
+}
+
+func TestChecksumStable(t *testing.T) {
+	require := require.New(t)
+
+	require.NoError(fixtures.Init())
+	defer func() {
+		require.NoError(fixtures.Clean())
+	}()
+
+	pool := NewRepositoryPool(cache.DefaultMaxSize)
+
+	for i, f := range fixtures.ByTag("worktree") {
+		path := f.Worktree().Root()
+		require.NoError(pool.AddGitWithID(fmt.Sprintf("repo_%d", i), path))
+	}
+
+	c := &checksumable{pool}
+
+	for i := 0; i < 100; i++ {
+		checksum, err := c.Checksum()
+		require.NoError(err)
+		require.Equal("mGPoKCyOIkXX4reGe1vTBPIOg2E=", checksum)
+	}
 }

